### ISSUE TITLE
why is `UnsoldAllocation.unlock` marked `payable`?

### ISSUE THREAD

**Issue by SCBuergel** [twentythirty/PillarToken#2]
I noticed that the `unlock` function in [UnsoldAllocation.sol](https://github.com/twentythirty/PillarToken/blob/master/contracts/UnsoldAllocation.sol#L38) is marked as `payable`. If I see correctly, the Ether send to these contracts will be lost as there is no way to retrieve them.

**Comment by ch4r10t33r** [comment]
SCBuergel - 

1) The UnsoldAllocation smart contract is an internal use only. This contract is used for icing unsold tokens in ice for a period specified. 
2) Calls to unlock function will throw as long as the specified lock-in period hasn't expired. So any ether you sent to this will be rejected and only gas would be consumed.

**Comment by SCBuergel** [comment]
> The UnsoldAllocation smart contract is an internal use only

No, it is marked as `external` and as such can be used by anyone and any contract out there.

> Calls to unlock function will throw as long as the specified lock-in period hasn't expired

Correct, that limits the problem to the eternity that follows the lock-in period.

**Comment by ch4r10t33r** [comment]
By internal use. I mean PillarProject team use only.

**Comment by SCBuergel** [comment]
No, anyone can call that function. But you're right, it would have been good to limit to `ownerOnly` or think if that is required in the first place at all. Many desperate people not getting into ICOs are sending to random addresses or calling random functions they find on Etherscan. We have seen that for almost every major ICO. Hence it is best practice to limit such incidents as much as possible and/or at least provide a method to return these Ether instead of just burning them forever.

**Comment by ch4r10t33r** [comment]
SCBuergel - I meant to say was that the function was supposed to be used only for internal/team only purpose however we didn't add the onlyOwner keyword for this. These contracts weren't audited as part of the ICO audit as they weren't customer facing. But point taken. Thanks.

**Comment by predicador37** [comment]
Hi @SCBuergel and @zincoshine :
If you use the ownerOnly modifier in the unlock() function of UnsoldAllocation, then no other but the PillarToken Contract could unlock the tokens, am I wrong? The owner would be the PillarToken since it's created by it...
